crates/publish/src/expander.rs

```diff
@@ -113,6 +113,24 @@ fn bindle_component_manifest(
                 "This version of Spin can't publish components whose sources are already bindles"
             )
         }
+        local_schema::RawModuleSource::Url(us) => match us.digest.split_once(':') {
+            None => anyhow::bail!(
+                "Invalid digest '{}' for '{}': must be sha256:...",
+                us.digest,
+                local.id
+            ),
+            Some((format, text)) => {
+                if format == "sha256" {
+                    text.to_owned()
+                } else {
+                    anyhow::bail!(
+                        "Invalid digest format '{}' for '{}': must be sha256",
+                        format,
+                        local.id
+                    )
+                }
+            }
```

Comment:
We really ought to share logic between `loader` and `publish`...

Reply:
Totally agree.  There's a _lot_ of tidying to do here, some of which will slot in more neatly with the improvements @etehtsea already has in progress.